plugins/include/sockets.inc

```diff
@@ -24,43 +21,159 @@
+ *
+ * @param _socket    Socket descriptor
+ * @param _data      Array to save the data
+ * @param _length    Length of the array
+ *
+ * @return           Ammount of bytes received
+ *                   0 if the peer closed the connection
+ *                   -1 on failure
+ */
 native socket_recv(_socket, _data[], _length);
 
-/* Sends data to the Socket */
-
+/**
+ * Sends data.
+ *
+ * @note The ammount of bytes than you end up sending can be less than expected, you're responsible of sending the rest later
+ *
+ * @param _socket    Socket descriptor
+ * @param _data      Array with the data to send
+ * @param _length    Length of the array
+ *
+ * @return           Ammount of bytes sent
+ *                   -1 on failure
+ */
 native socket_send(_socket, const _data[], _length);
 
-/* Same as socket_send but Data can contain null bytes */
-
+/**
+ * Sends data that can contain null bytes.
+ *
+ * @note The ammount of bytes than you end up sending can be less than expected, you're responsible of sending the rest later
+ *
+ * @param _socket    Socket descriptor
+ * @param _data      Array with the data to send
+ * @param _length    Length of the array
+ *
+ * @return           Ammount of bytes sent
+ *                   -1 on failure
+ */
 native socket_send2(_socket, const _data[], _length);
 
-/* This function will return true if the state (buffer content) have changed within the last recieve or
-* the timeout, where timeout is a value in µSeconds, (1 sec =1000000 µsec).
-* Use to check if new data is in your socket. */
-
-native socket_change(_socket, _timeout=100000);
+/**
+ * Check if a socket is marked as readable.
+ *
+ * @note You can use this function to make sure there's something on the socket and avoiding a blocking call
+ * @note Set _timeout to 0 avoid blocking the call
+ * @note A socket will become readable if there's any data or an EOF
+ *
+ * @param _socket    Socket descriptor
+ * @param _timeout   Ammount of time to block the call waiting for the socket to be marked as readable or for the timeout to expire, in µSeconds (1 sec = 1000000 µsec)
+ *
+ * @return           1 if the socket is marked as readable
+ *                   -1 otherwise
+ */
+native socket_change(_socket, _timeout = 100000);
+
+/**
+ * Check if a socket is marked as readable.
+ *
+ * @note You can use this function to make sure there's something on the socket and avoiding a blocking call
+ * @note Set _timeout to 0 avoid blocking the call
+ * @note A socket will become readable if there's any data or an EOF
+ *
+ * @param _socket    Socket descriptor
+ * @param _timeout   Ammount of time to block the call waiting for the socket to be marked as readable or for the timeout to expire, in µSeconds (1 sec = 1000000 µsec)
+ *
+ * @return           1 if the socket is marked as readable
+ *                   -1 otherwise
+ */
+native socket_is_readable(_socket, _timeout = 100000);
```

Comment:
I don't know. Kind of silly, but for the sake of consistency and readability, it could make sense.